frontend/src/schemas/SchemaList.tsx

```diff
@@ -1,35 +1,75 @@
-import { Button, List, ListItem, ListItemText } from '@mui/material'
-import { useGetSchemas } from 'actions/schema'
-import { useMemo } from 'react'
+import { Button, List, Stack, Typography } from '@mui/material'
+import { deleteSchema, putSchema, useGetSchemas } from 'actions/schema'
+import { useMemo, useState } from 'react'
+import ConfirmationDialogue from 'src/common/ConfirmationDialogue'
+import EmptyBlob from 'src/common/EmptyBlob'
 import Loading from 'src/common/Loading'
 import MessageAlert from 'src/MessageAlert'
+import { SchemaInterface } from 'types/types'
+import { getErrorMessage } from 'utils/fetcher'
 
 interface SchemaDisplayProps {
   schemaKind: string
 }
 
 export default function SchemaList({ schemaKind }: SchemaDisplayProps) {
-  const { schemas, isSchemasLoading, isSchemasError } = useGetSchemas(schemaKind)
-
-  const schemaList = useMemo(
-    () =>
-      schemas.map((schema, index) => (
-        <ListItem
-          key={schema.id}
-          divider={index + 1 !== schemas.length}
-          disableGutters
-          secondaryAction={
-            <>
-              <Button disabled>Edit</Button>
-              <Button disabled>{schema.active ? 'Set as inactive' : 'Mark as active'}</Button>
-            </>
-          }
-        >
-          <ListItemText primary={schema.name} sx={{ maxWidth: 'fit-content', pr: 20 }} />
-        </ListItem>
-      )),
-    [schemas],
-  )
+  const { schemas, isSchemasLoading, isSchemasError, mutateSchemas } = useGetSchemas(schemaKind)
+
+  const [errorMessage, setErrorMessage] = useState('')
+  const [open, setOpen] = useState(false)
+  const [schemaToBeDeleted, setSchemaToBeDeleted] = useState('')
+
+  const schemaList = useMemo(() => {
```

Comment:
memo hooks shouldn't contain side effects. I'd recommend moving most of this logic into a new component and using the memo to only handle managing a list of those components.